src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs

```diff
@@ -442,8 +387,7 @@ private async Task ConnectAsync(CancellationToken token)
             {
                 await BackOff();
                 _stateSubject.OnNext(GraphQLWebsocketConnectionState.Connecting);
-                Debug.WriteLine($"opening websocket {_clientWebSocket.GetHashCode()} (thread {Thread.CurrentThread.ManagedThreadId})");
-                await _clientWebSocket.ConnectAsync(_webSocketUri, token);
+                _clientWebSocket = await _webSocketFactory.ConnectAsync(_webSocketUri, token);
```

Comment:
@rose-a @andreyleskov regarding 4. Tests stability:

I recommend you **ALWAYS** use `ConfigureAwait(false)` in library code especially when using XUnit package as a test framework because of [`Xunit.Sdk.MaxConcurrencySyncContext`](https://github.com/xunit/xunit/blob/main/src/xunit.v3.core/Sdk/MaxConcurrencySyncContext.cs). In my case `maximumConcurrencyLevel` was 6 and some tests randomly failed. So please look through entire codebase and add `ConfigureAwait(false)` to awaitable calls. Perhaps it will help.

Reply:
Thanks a lot for the comment, it makes sense from my point of view. This PR is following the existing await style and focuses on just WebSockets. I assume repository owners prefer to do repository-wide changes as await style in a dedicated PR.

Reply:
im on to this in in #370...